### PR TITLE
planner: Fix the 'Unknown column' error when select from view in another database

### LINKED ISSUE #15611: UCP: unexpected error happened when select dbName.viewName.col from another database
### 1. What did you do?
``` sql
use test;
create table t(a int);
use mysql;
create view v as select * from test.t;
use test;
select mysql.v.a from mysql.v;
```

### 2. What did you expect to see?
no error happens

### 3. What did you see instead?
`ERROR 1054 (42S22): Unknown column 'mysql.v.a' in 'field list'`

## Score

* 50

## Mentor(s)

Contact the mentors: **#tidb-challenge-program** channel in [TiDB Community](https://join.slack.com/t/tidbcommunity/shared_invite/enQtNzc0MzI4ODExMDc4LWYwYmIzMjZkYzJiNDUxMmZlN2FiMGJkZjAyMzQ5NGU0NGY0NzI3NTYwMjAyNGQ1N2I2ZjAxNzc1OGUwYWM0NzE)  Slack Workspace

## Recommended Skills

* Golang Programming

## Learning Materials

* Chinese: [TiDB 精选技术讲解文章](https://github.com/pingcap/presentations/blob/master/hackathon-2019/reference-document-of-hackathon-2019.md)
* English: [Awesome-Database-Learning](https://github.com/pingcap/awesome-database-learning)

### PR DESCRIPTION
**Pull request by longfangsong**
### What problem does this PR solve?

fix #15611

Issue Number: close #15611

### What is changed and how it works?

What's Changed:

Changing [this line](https://github.com/pingcap/tidb/blob/master/planner/core/logical_plan_builder.go#L2930) to `DBName: dbName`

Just as @XuHuaiyu mentioned.

### Related changes

NA

### Release note 

NA

Unit test is already added.

### CODE REVIEW

**Comment by zz-jason** [comment]
Hi @longfangsong, thanks for your contribution, please add a UT for this bugfix.

**Comment by longfangsong** [comment]
> Hi @longfangsong, thanks for your contribution, please add a UT for this bugfix.

I'm sorry that I'm new in tidb developing, so I'm not sure whether I put the test in the right place.

If I'm wrong, could you please tell me where should I put it in.

File: planner/core/logical_plan_builder.go [hunk 1]
```diff
@@ -2927,7 +2927,7 @@ func (b *PlanBuilder) buildProjUponView(ctx context.Context, dbName model.CIStr,
 			OrigTblName: name.OrigTblName,
 			ColName:     columnInfo[i].Name,
 			OrigColName: origColName,
-			DBName:      name.DBName,
+			DBName:      dbName,
```
Comment (zz-jason) [inline planner/core/logical_plan_builder.go]: @XuHuaiyu Do you have any advise about how to test this bugfix?

**Comment by zz-jason** [comment]
> > Hi @longfangsong, thanks for your contribution, please add a UT for this bugfix.
> 
> I'm sorry that I'm new in tidb developing, so I'm not sure whether I put the test in the right place.
> 
> If I'm wrong, could you please tell me where should I put it in.

Don't worry, we'll help you. 😄

**Comment by XuHuaiyu** [comment]
@longfangsong 
The test function looks good to me. ^_^
Can we remove the `[WIP]` in the title now?
And, please modify the title to a more meaningful sentence about `what you do in this PR`.

**Pull request merged**